webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ExpectedReportsJdbcRepository.scala

```diff
@@ -206,10 +206,30 @@ class FindExpectedReportsJdbcRepository(
     }
   }
 
+  /**
+   * Return node ids associated to the directive (based on expectedreports (the one still pending)) for this Directive,
+   * only limited on the nodeIds in parameter (used when cache is incomplete)
+   */
+  override def findCurrentNodeIdsForDirective(directiveId: DirectiveId, nodeIds: Set[NodeId]): IOResult[Set[NodeId]]                              = {
+    if (nodeIds.isEmpty) Set.empty[NodeId].succeed
+    else {
+      transactIOResult(s"Error when getting nodes for directive '${directiveId.serialize}' from expected reports")(xa => sql"""
+        select distinct nodeid from nodeconfigurations
+        where enddate is null and configuration like ${"%" + directiveId.serialize + "%"}
+        and nodeid in (${nodeIds.map(id => s"'${id}'").mkString(",")})
+      """.query[NodeId].to[Set].transact(xa))
+    }
+  }
+  override def findCurrentNodeIdsForDirective(directiveId: DirectiveId):                       IOResult[Set[NodeId]]                              = {
+    transactIOResult(s"Error when getting nodes for directive '${directiveId.serialize}' from expected reports")(xa => sql"""
+      select distinct nodeid from nodeconfigurations
+      where enddate is null and configuration like ${"%" + directiveId.serialize + "%"}
+    """.query[NodeId].to[Set].transact(xa))
+  }
```

Comment:
can you factor out the common bit? I think I saw you use something for and part somewhere